<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1037</width>
    <height>767</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Quorum Sensing Simulation</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_3">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,2,1">
      <item row="0" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QToolButton" name="playButton">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="colonyResourceFile.qrc">
            <normaloff>:/icons/realistik_player_play_48.png</normaloff>:/icons/realistik_player_play_48.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>48</width>
            <height>48</height>
           </size>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="pauseButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="colonyResourceFile.qrc">
            <normaloff>:/icons/realistik_player_pause_48.png</normaloff>:/icons/realistik_player_pause_48.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>48</width>
            <height>48</height>
           </size>
          </property>
          <property name="autoRaise">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="simulationProgressBar">
          <property name="maximumSize">
           <size>
            <width>300</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QGridLayout" name="gridLayout">
        <property name="leftMargin">
         <number>5</number>
        </property>
        <property name="rightMargin">
         <number>10</number>
        </property>
        <item row="1" column="0">
         <widget class="QLabel" name="NCells">
          <property name="font">
           <font>
            <family>Sans Serif</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>NCells</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="NCellsNum">
          <property name="font">
           <font>
            <family>Monospace</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="time">
          <property name="font">
           <font>
            <family>Sans Serif</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>time</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="timeNum">
          <property name="font">
           <font>
            <family>Monospace</family>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="iTraj">
          <property name="text">
           <string>trajectory #</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="iTrajNum">
          <property name="font">
           <font>
            <family>Monospace</family>
           </font>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="iParameterSet">
          <property name="text">
           <string>Parameter set #</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="iParameterSetNum">
          <property name="font">
           <font>
            <family>Monospace</family>
           </font>
          </property>
          <property name="text">
           <string>0</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>700</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>Cell Color Code</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <property name="topMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="2,1,1">
             <item>
              <widget class="QGraphicsView" name="colorCodeGraphicsView">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>30</width>
                 <height>30</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>200</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>150</width>
                 <height>30</height>
                </size>
               </property>
               <property name="verticalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOff</enum>
               </property>
               <property name="horizontalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOff</enum>
               </property>
               <property name="interactive">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="speciesColorMaxValueSpinBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
               <property name="accelerated">
                <bool>true</bool>
               </property>
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="maximum">
                <double>10000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>10.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="speciesColorComboBox">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
          <zorder></zorder>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>700</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>Background Color Code</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="topMargin">
            <number>5</number>
           </property>
           <property name="bottomMargin">
            <number>5</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="2,1,1">
             <item>
              <widget class="QGraphicsView" name="colorCodeBackgroundGraphicsView">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>30</width>
                 <height>30</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>200</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>150</width>
                 <height>30</height>
                </size>
               </property>
               <property name="verticalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOff</enum>
               </property>
               <property name="horizontalScrollBarPolicy">
                <enum>Qt::ScrollBarAlwaysOff</enum>
               </property>
               <property name="interactive">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="speciesBackgroundColorMaxValueSpinBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
               <property name="decimals">
                <number>0</number>
               </property>
               <property name="maximum">
                <double>10000.000000000000000</double>
               </property>
               <property name="singleStep">
                <double>10.000000000000000</double>
               </property>
               <property name="value">
                <double>100.000000000000000</double>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="speciesBackgroundColorComboBox">
               <property name="minimumSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
               <property name="baseSize">
                <size>
                 <width>60</width>
                 <height>30</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="font">
         <font>
          <family>Sans Serif</family>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Global Parameters</string>
        </property>
        <property name="indent">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QLabel" name="label_3">
        <property name="font">
         <font>
          <family>Sans Serif</family>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Cell State</string>
        </property>
        <property name="indent">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QTableView" name="globalParameterTableView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="baseSize">
         <size>
          <width>200</width>
          <height>500</height>
         </size>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <attribute name="horizontalHeaderVisible">
         <bool>false</bool>
        </attribute>
        <attribute name="verticalHeaderVisible">
         <bool>false</bool>
        </attribute>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QGraphicsView" name="colonyView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>700</width>
          <height>700</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="toolTip">
         <string/>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
        <property name="lineWidth">
         <number>1</number>
        </property>
        <property name="renderHints">
         <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
        </property>
        <property name="cacheMode">
         <set>QGraphicsView::CacheBackground</set>
        </property>
        <property name="viewportUpdateMode">
         <enum>QGraphicsView::SmartViewportUpdate</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QTableView" name="cellStateTableView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="baseSize">
         <size>
          <width>200</width>
          <height>500</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="font">
           <font>
            <family>Sans Serif</family>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Zoom</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="zoomSlider">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>440</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="colonyResourceFile.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateSimulationProgressBar(int)</signal>
   <receiver>simulationProgressBar</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>268</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>63</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>playSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>32</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>46</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateTrajectoryCounter(QString)</signal>
   <receiver>iTrajNum</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>132</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateNCells(int)</signal>
   <receiver>NCellsNum</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>167</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>137</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateTimeSimulation(QString)</signal>
   <receiver>timeNum</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>updateParameterSetCounter(QString)</signal>
   <receiver>iParameterSetNum</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>216</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zoomSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>on_zoomSlider_valueChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>732</y>
    </hint>
    <hint type="destinationlabel">
     <x>230</x>
     <y>548</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speciesColorComboBox</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSpeciesCellColorIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>532</x>
     <y>36</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speciesColorMaxValueSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSpeciesCellColorMaxValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>488</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speciesBackgroundColorMaxValueSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSpeciesBackgroundColorMaxValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>755</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>speciesBackgroundColorComboBox</sender>
   <signal>activated(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setSpeciesBackgroundColorIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>723</x>
     <y>204</y>
    </hint>
    <hint type="destinationlabel">
     <x>836</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>pauseSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>283</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>updateNCells(int)</signal>
  <signal>updateTimeSimulation(QString)</signal>
  <signal>updateTotalTimeSimulation(QString)</signal>
  <signal>updateTrajectoryCounter(QString)</signal>
  <signal>updateParameterSetCounter(QString)</signal>
  <signal>updateSimulationProgressBar(int)</signal>
  <signal>updateGlobalParameterTable()</signal>
  <slot>playSimulation()</slot>
  <slot>resetSimulation()</slot>
  <slot>on_zoomSlider_valueChanged(int)</slot>
  <slot>setSpeciesCellColorIndex(int)</slot>
  <slot>setSpeciesCellColorMaxValue(double)</slot>
  <slot>setSpeciesBackgroundColorMaxValue(double)</slot>
  <slot>setSpeciesBackgroundColorIndex(int)</slot>
  <slot>updateSpeciesColorCodeView()</slot>
  <slot>pauseSimulation()</slot>
  <slot>openOpenGLViewer()</slot>
 </slots>
</ui>
